plugins/installer.plugin

```diff
@@ -53,22 +76,104 @@ finalize_deployment_installer()
   cp /usr/lib/molior-deploy/molior-deploy.sh.inc $initrd/scripts/
 
   if [ "$fs_has_luks" -eq 1 ]; then
+    # FIXME: use manual_add_modules in hooks file ?
     echo "copy_exec /sbin/cryptsetup /sbin" >> $initrd/hooks/installer
     echo dm_crypt >> $initrd/modules
   fi
 
+  if [ "$INSTALLER_BOOTTYPE" = "efi" ]; then
+    cat >$initrd/hooks/mkfs.vfat <<EOF
+#!/bin/bash
+
+PREREQ=""
+
+prereqs()
+{
+        echo "\$PREREQ"
+}
+
+case \$1 in
+prereqs)
+        prereqs
+        exit 0
+        ;;
+esac
+
+. /usr/share/initramfs-tools/hook-functions
+
+manual_add_modules efivars
+copy_exec      /sbin/mkfs.fat  /sbin/
+copy_file link /sbin/mkfs.vfat /sbin/
+EOF
+    chmod +x $initrd/hooks/mkfs.vfat
+  fi
+
   initrd=`readlink $WORK_DIR/instroot/initrd.img | xargs basename`
   chroot $WORK_DIR/instroot update-initramfs -u >/dev/null
   if [ $? -ne 0 ]; then
     log_error "Error updating installer initrd"
   fi
+
+  if [ "$INSTALLER_BOOTTYPE" = "efi" ]; then
+    mkdir -p $WORK_DIR/iso/boot/grub/$GRUBEFITYPE/
+    cp $WORK_DIR/instroot/usr/lib/grub/$GRUBEFITYPE/* $WORK_DIR/iso/boot/grub/$GRUBEFITYPE/
+    cat >$WORK_DIR/iso/boot/grub/grub.cfg <<EOF
+search --file --set=root /image.cpio.xz
+insmod all_video
+linux /vmlinuz quiet
+initrd /initrd.img
+boot
+EOF
+    chroot $WORK_DIR/instroot grub-mkimage -O $GRUBEFITYPE -p /boot/grub -o /boot/$EFIFILE \
+                                           part_msdos part_gpt fat iso9660 search
+    if [ $? -ne 0 ]; then
+      log_error "Error creating grub image"
+    fi
+    mkdir -p $WORK_DIR/iso/efi/boot/
+    cp $WORK_DIR/instroot/boot/$EFIFILE $WORK_DIR/iso/efi/boot/
+
+    if [ "$ARCH" = "arm64" ]; then
+      mkdir $WORK_DIR/di
```

Comment:
di ?

Reply:
debian-installer

Reply:
Rename it to `debian-installer` please :)

Reply:
done